src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -315,7 +315,8 @@ Connection newConnection() throws Exception
       }
       catch (Exception e) {
          lastConnectionFailure.set(e);
-         quietlyCloseConnection(connection, "(Failed to create/set connection)");
+         String reason = e.getMessage();
+         quietlyCloseConnection(connection, reason != null ? reason : "(Failed to create/set connection)");
```

Comment:
it is better to know 'actual reason', this text is more important as there can be many reason and is used for logging purpose to help debug/trace problem